### ISSUE TITLE
Login with CAS gives me : "Username already exists."

### ISSUE THREAD

**Issue by anthony-o** [RocketChat/Rocket.Chat#3379]
Your Rocket.Chat version: 0.31.0

I've enabled CAS on my Rocket.Chat. This instance was created with v0.21.0 and my user was the first one to be created and the admin.

If I try to login with CAS, I've got "Username already exists." error message and I stay on login screen.
But other users don't have that problem: they can correctly manage to login with the CAS button.

**Comment by corecache** [comment]
This seems to happen when RocketChat accounts were created before 6b5565a36e0e9b3a4f5ac54e67a0e0fc77280460 
The way how user are identified/looked up changed with that patch.

Short fix if you can live with the consequences for now:
remove the affected users by RocketChat admin account, the user will be recreated correctly when authenticating next time with CAS sso.

**Comment by anthony-o** [comment]
But if I remove this user, all his posts will be deleted?
Is there any way I can go throughout this bug by accessing the database and modifying some fields of some entity?

**Comment by corecache** [comment]
That's the downside of deleting the user.

Of course you can edit the user directly in the database by using `meteor shell` and update the user record:

`var username = 'yourusername';`
`var myuser = Meteor.users.findOne({ 'username': username });`
`Meteor.users.update(myuser, { $set: { services: { cas: { external_id: username } } } });`

**Comment by anthony-o** [comment]
Great I accessed the mongodb server directly (because I was unable to find `meteor` binary on the rocketchat/rocket.chat:0.41.0 docker image) and added the `cas` field to the `services` and it worked, thanks !

**Comment by corecache** [comment]
For future reference, use something with docker like this:

`docker exec -ti rocketchat_rocketchat_1 bash`
`root@61dec02b0979:/app# meteor shell`

**Comment by anthony-o** [comment]
After being connected to the rocketchat container (via `docker exec -it rocketchat_rocketchat_1 bash`), I'm still unable to execute that command:
```
rocketchat@3e128192d649:/app/bundle$ cd ..
rocketchat@3e128192d649:/app$ meteor shell
bash: meteor: command not found
```

**Comment by corecache** [comment]
@anthony-o This issue can be closed?

**Comment by domenicocosta** [comment]
Hi,

i have installed in my local machine a server RC and i have to sincronize user via CAS, but i am obbliged to leave the checkbox "Registration with Authentication Services"  to true, because in your record user exist the services json for cas.

"services": {
		"cas": {
			"external_id": "******",
			"version": 1,
			"attrs": {}
		},
		"resume": {
			"loginTokens": [{
				"when": ISODate("2019-02-13T20:12:08.473Z"),
				"hashedToken": "JIfnasKSEOzEh1UWhUAmsIgaEOo9R2lFX1ijGBQuF2Q="
			}]
		}
	},

For my purpose i create the user via rest api and next to access in rocket chat via CAS, but with the checkbox setting in true i have the error "User Exist...", it's possible to carry out an update to the record user with the services cas and check the first via CAS and the next via Password.

It's presumibile resolve this issue in this mode?

Thanks

Best regards

**Comment by bseclier** [comment]
Hello, I don't understand why this issue is closed, I still have the problem here (v3.4.1).
Was someone be able to create a user with the API and, then, connect with this user thanks to CAS without have the "username already exists" ?

Thanks for your help.

**Comment by bseclier** [comment]
A fix was added to the fix, it works perfectly now (3.4.2).
Thanks.
